### DISCUSSION TITLE
Is there ability to see, change and delete selected `/file` ?

### DISCUSSION THREAD

**Discussion by antosha417** [olimorris/codecompanion.nvim#349] [Q&A]
Hey! First of all many thanks for the great plugin, I just recently started to use it, but I love it! 
It is just what I was looking for after using cursor ai for some time.

My question is about `/file` command. When I choose the file I don't see any indication of it in the chat window. Would be great to have it. Some file icon and a file name should be more than enough I think. Should I enable it in some way or it is not supported yet?

Also the ability to change or delete selected file if I choose the wrong file would be awesome.

**Comment by antosha417** [comment]
currently I can see the file with `gd` keybinding, but it would be much more easier to have some indication in the chat

**Comment by olimorris** [comment]
You should be getting a notification via `vim.notify` that the file has been added. This was discussed in https://github.com/olimorris/codecompanion.nvim/discussions/309 a couple of weeks ago.

No plans to enable this for now.

**Reply by antosha417** [reply]
I'm not talking about the whole content of the file, only file name would be fine. Would you at least consider having a config for this?

**Reply by olimorris** [reply]
To clarify, you're not seeing any notification that a file has been added to the buffer?

Also, how long are you typically having chat buffers open for? Are they persisting across a session for days?

It's not as simple as having a config option. There's the UI considerations for this as well. How do we display this information to the users. How do we then strip it out before sending to an LLM. The poll in #309 suggests most users don't actually need this so I need to balance those considerations.

**Reply by antosha417** [reply]
Yeah, currently no notifications, but this is probably something on my side. 

Buffers are not persisting across sessions, but rather my, sessions are persisting for a days. 

I get the problem in UI considerations. But we do actually send file to the llm. 
If I understand correctly currently when we include the file in the chat we have 
```lua
    content = "Buffer ID: 69\n...",
    opts = {
      tag = "variable",
      visible = false
    },
    ...
```

I think for me would be ideal if we put something else in the `opts` that would indicate that this is a `/file` variable and would just show to the user file icon and file name there. Instead of hiding this entirely. I think this could be configurable. We could even allow custom show handlers for the variables so people could override it as they wish and maybe ever see the whole file content there. 


Regarding the poll in https://github.com/olimorris/codecompanion.nvim/discussions/309 I'm not entirely sure it represents most of the users, since there are only 7 votes (one I made couple hours ago). I think this awesome plugin has many more users 


If we agree on some approach here I might help implementing it if you want.
